# ML-02 — Research Question and Provisional Lane

[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/damlablgc/flyrankinternship/blob/main/work/notebooks/w01_research_question.ipynb)

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

I'm picking **Lane 2 — Refresh/Content Opportunity Scoring**. My random forest model beat the baseline rules (0.750 AUC vs 0.627 AUC, precision@50 went from 0.240 to 0.740) which shows a learned model can actually do better than a simple rule here. I also found that comparison articles get lower CTR than other content types even at the same position tier, a signal i can fold into my scoring later. This lane feels worth it because content teams can't manually review every page, so a ranked "review this first" list is genuinely useful.

## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

My work improves the decision of which pages to prioritize for content refresh instead of the old "oldest page first" approach, the **content team** (Who acts on it) gets a ranked list based on real signals (declining traffic, low visibility, thin content, etc.) so their limited time goes to pages that are actually worth the effort.

Cost of a wrong call:

False positive, the team wastes time editing a page that didn't need it.

False negative, that page keeps losing traffic unnoticed and the opportunity to fix it early is lost.

## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [7]:
baseline = results["models"]["decision_tree"]  # not used here, just checking structure
rf = results["models"]["random_forest"]
baseline_metrics = results["baseline"]

print("=== Numbers supporting Lane 2 ===")
print(f"1. Pages currently declining: {declining_count} / {len(df)} ({declining_pct:.1f}%)")
print(f"2. Baseline rule ROC AUC: {baseline_metrics['baseline_roc_auc']:.3f} "
      f"vs Random Forest ROC AUC: {rf['roc_auc']:.3f}")
print(f"3. Baseline Precision@50: {baseline_metrics['baseline_precision_at_50']:.2f} "
      f"vs Random Forest Precision@50: {rf['precision_at_50']:.2f}")
print(f"\nTop feature driving the model: {results['best_model']['feature_importance_top'][0]['feature']}")

=== Numbers supporting Lane 2 ===
1. Pages currently declining: 16262 / 30000 (54.2%)
2. Baseline rule ROC AUC: 0.627 vs Random Forest ROC AUC: 0.750
3. Baseline Precision@50: 0.24 vs Random Forest Precision@50: 0.74

Top feature driving the model: days_with_impressions


Out of 30,000 pages in the starter slice, **16,262 (54.2%) are currently labeled "declining"** — so this isn't a rare edge case, it's more than half the inventory, which makes prioritization genuinely necessary.

When I compared the baseline rule against my trained models: the baseline rule scored **0.627 ROC AUC / 0.24 precision@50**, while my random forest scored **0.750 ROC AUC / 0.74 precision@50**. In practical terms, out of a top-50 review queue, the baseline gets ~12 pages right, while the model gets ~37 right — roughly triple the useful hits for the same review effort.

The single strongest feature in the model was `days_with_impressions` (how many of the last 90 days the page actually got any visibility at all) — which makes sense as an early-warning signal for a page quietly losing exposure.

## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

**What I can say:**
- 16,262 out of 30,000 pages (54.2%) were observed as currently declining in this slice.
- The random forest ranking outperformed the baseline rule (0.750 vs 0.627 ROC AUC, 0.74 vs 0.24 precision@50) on this data.
- This ranking can act as a decision-support guide helping the content team prioritize which pages to review before they decline further, not a guarantee of what will happen to any single page.

**What I can never say:**
- That refreshing a page *causes* it to recover, that would need a real experiment (A/B test), not just this ranking.
- That I've proven anything about how Google's algorithm actually works.
- That `is_declining_label` predicts the future, it's a proxy calculated from the current 90-day window, not an actual future outcome.

## Self-check

Before you submit, confirm each line honestly:

- [x] Every section above is filled — markdown thinking AND the code that backs it
- [x] The notebook runs top to bottom with no errors (Runtime → Run all)
- [ ] No client names, URLs, or private queries anywhere
- [ ] My claims use careful words: observed, measured, directional, decision-support
- [ ] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.